### ISSUE TITLE
__red() and probably __green() fail for url encoded text

### ISSUE THREAD

**Issue by dkerr64** [Neilpang/acme.sh#549]
When issuing an error using _err() function if the string passed in is url encoded then the __red() function throws an error.

for example...
`_err "user name%40kerrdomain.tld failed"`
the % caused printf to try and parse what follows.... debug log shows...

```
[Wed Jan 18 12:39:52 EST 2017] /stat/etc/acme/acme.sh: line 83: printf: `k': invalid format character
user name
```
As you can see printout stops at the % sign.

Thanks
David

**Comment by Neilpang** [comment]
please let me which line cause this error ?

**Comment by dkerr64** [comment]
The error is at line 83 as per the error in my original report.  I am working on a DNS api plugin for FreeDNS (which I will submit for you to consider including with acme.sh soon) and needed to print out an error message if login to FreeDNS fails,  I included in that message the FreeDNS userid which was an email address.  The @ in the address was encoded as %40 and that caused this issue.

I have solved the problem by not using the URL encoded string in the error message.  So for now this is not a problem and can be deferred if you like.

Thanks
David
